frontend/app/dashboard/Canvas.tsx

```diff
@@ -826,13 +796,15 @@ export function Canvas({
 }
 
 function getPrimaryColor(id: UniqueIdentifier) {
-  const hash = simpleHash(String(id).split('|')[1].slice(0, 3));
-  return PRIMARY_COLOR_LIST[hash];
+  const dept = String(id).split('|')[1].slice(0, 3).toUpperCase();
+  const gradient = getDepartmentGradient(dept, 90);
+  return gradient.match(/#[0-9a-fA-F]{6}/)?.[0] ?? '#000000'; // Extract the first color
 }
 
 function getSecondaryColor(id: UniqueIdentifier) {
-  const hash = simpleHash(String(id).split('|')[1].slice(0, 3));
-  return SECONDARY_COLOR_LIST[hash];
+  const dept = String(id).split('|')[1].slice(0, 3).toUpperCase();
+  const gradient = getDepartmentGradient(dept, 90);
+  return gradient.match(/#[0-9a-fA-F]{6}/g)?.[1] ?? '#FFFFFF'; // Extract the second color
```

Comment:
Should this regex have a g flag? Or maybe is there an easier way to extract the color without using regex?